src/Logic.hs

```diff
@@ -597,13 +597,15 @@ handleBuildStatusChanged buildSha newStatus state = pure $
 -- * Statuses 'BuildSuceeded' and 'BuildFailed' are not
 --   superseded by other statuses
 --
+-- * We ignore changes of just the URL.
+--
 -- This is used in 'handleBuildStatusChanged`.
 supersedes :: BuildStatus -> BuildStatus -> Bool
-newStatus       `supersedes` oldStatus       | newStatus == oldStatus = False
-(BuildFailed _) `supersedes` (BuildFailed _) = True
-_               `supersedes` (BuildFailed _) = False
-_               `supersedes` BuildSucceeded  = False
-_               `supersedes` _               = True
+newStatus        `supersedes` oldStatus        | newStatus == oldStatus = False
+(BuildStarted _) `supersedes` (BuildStarted _) = False -- url change, ignore.
```

Comment:
Perhaps a more robust thing to do here is change the guard on the first case from `newStatus == oldStatus` to something like `sameStatusType newStatus oldStatus` where `sameStatusType` is URL-insensitive equality, i.e.

```haskell
sameStatusType BuildStarted{} BuildStarted{} = True
sameStatusType BuildFailed{}  BuildFailed{}  = True
sameStatusType x              y              = x == y
```

Thoughts?

Reply:
@alex-mckenna nice trick!  Pattern matching using an empty record notation has never occurred to me before.